appengine/pubsub/spec/pubsub_spec.rb

```diff
@@ -47,6 +50,41 @@ def app
     expect(last_response.status).to eq 303
   end
 
+  it "accepts a push" do
+    post "/pubsub/push?token=#{ENV["PUBSUB_VERIFICATION_TOKEN"]}",
+         JSON.generate({"message" => { "data" => Base64.encode64("A Message") }})
+
+    expect(last_response.status).to eq 200
+  end
+
+  it "accepts an authenticated push" do
+    public_cert_str = File.read "spec/fixtures/public_cert.pem"
+    key = OpenSSL::X509::Certificate.new(public_cert_str).public_key
+    key_info = Google::Auth::IDTokens::KeyInfo.new id: "test-key", key: key, algorithm: "RS256"
+    key_source = Google::Auth::IDTokens::StaticKeySource.new key_info
+    Google::Auth::IDTokens.instance_variable_set(:@oidc_key_source, key_source)
```

Comment:
One thing I forgot to say. In an ensure block after the test completes, reset this by calling `Google::Auth::IDTokens.forget_sources!` so it doesn't clobber any future tests that depend on IDTokens.

Reply:
Done, thanks!